src/product/dto/product-response.dto.ts

```diff
@@ -14,4 +14,20 @@ export class ProductResponseDto extends CreateProductDTO {
             type: product.type
         }
     }
+
+    static async response(product: Promise<Product>): Promise<ProductResponseDto> {
+        try {
+            const result = await product
```

Comment:
I think that you could learn more about async/await and Promise, i feel that you need more experience with thoses themes